runtime/resolvers/glob.go

```diff
@@ -372,26 +376,27 @@ func (r *globResolver) writeTempNDJSONFile(rows []map[string]any) (string, error
 }
 
 func readNDJSONFile(filePath string) ([]map[string]any, error) {
-	f, err := os.Open(filePath)
+	db, err := sqlx.Open("duckdb", "")
 	if err != nil {
 		return nil, err
 	}
-	defer f.Close()
+	defer db.Close()
 
-	dec := json.NewDecoder(f)
-	var rows []map[string]any
-	for {
-		var row map[string]any
-		if err := dec.Decode(&row); err != nil {
-			if errors.Is(err, io.EOF) {
-				break
-			}
+	// read filepath as ndjson
+	var data []map[string]any
+	rows, err := db.QueryxContext(context.Background(), fmt.Sprintf("SELECT * FROM read_ndjson_auto(%s)", safeSQLString(filePath)))
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		row := make(map[string]any)
+		if err := rows.MapScan(row); err != nil {
 			return nil, err
 		}
-		rows = append(rows, row)
+		data = append(data, row)
 	}
```

Comment:
Could this be moved to around line 341, so it just reads the rows directly instead of doing `COPY` to JSON and calling `readNDJSONFile`?  That removes the need for `read_ndjson_auto` to infer the schema for the output of `transform_sql`.